.github/CI/test.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash -e
+
+source .github/CI/env_setup.sh
+
+if [ "$BUILD_TYPE" = "Release" ]; then
```

Comment:
OK, I made the conditional also include the running of ctest.  Is this what we want, or should we put the conditional in the yaml?

Reply:
Having the conditional in the yaml would be nicer.

Reply:
I discussed this with Aurelien before the last update.  We thought that it made more sense to have all the logic in one place and that this would also cause the scripts to work the same whether or not they are being executed through the CI or in an interactive session.

Reply:
I find it strange to put such logic into helper files instead of having it in the file that was meant for that (aka. the yaml). But I might have a skewed view as all the other projects I'm involved with have the CI driving login in the yaml and the other scripts are just helpers.

Reply:
I think some people must find it nice to have everything in one file or in as few files as possible.  I do not find that to be nice to maintain in this case for two major reasons.  The first is that having one language or markup embedded in another unnecessarily complicates the process of understanding what the code is doing.  You have to keep track of the context of what you are looking at line-by-line in order to know if it is bash script of yaml.  You also have to be more careful of the escaping and referencing of variables because the yaml variables are referenced differently than in the shell script.  The shell script within the yaml also has to be unnecessarily indented to satisfy the requirements of both the yaml markup and the style of the shell script (this indentation could be different). I find it much cleaner to have one markup or language in a given file.  This way, someone looking at the much shorter yaml can see that it just iterates over the matric values and runs the shell scripts in sequence.  The shell scripts are divided up in order to encapsulate the functionality.  If you are interested in modifying the test or build or configuration process, then it is very clear where you would do that, and you don't have to worry nearly as much about affecting other parts of the CI because the context of the functionality is limited.